src/connector/src/parser/mod.rs

```diff
@@ -237,7 +239,7 @@ impl SourceStreamChunkRowWriter<'_> {
             .zip_eq(self.builders.iter_mut())
             .enumerate()
             .try_for_each(|(idx, (desc, builder))| -> Result<()> {
-                if desc.is_meta {
+                if desc.is_meta || desc.is_offset {
```

Comment:
I think it is hard to do. Since `is_meta` denotes another internal column, basically we need two flags to denote two internal columns and fill up the hole.

Reply:
We identify the semantics of the meta column by the column name and the `SourceMeta`. So I guess there won't be any confusion here?

Reply:
Refactor to a enum field `column_type`